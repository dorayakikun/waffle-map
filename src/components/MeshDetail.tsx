import { Table, Tbody, Tr, Td } from "@chakra-ui/react";
import * as React from "react";
import { Bounds, LatLng } from "../domain/calculateMesh";
import { round } from "../domain/roundPoint";

export type Props = {
  code: string;
  center: LatLng;
  bounds: Bounds;
};

type TableRowValue = {
  latLng: LatLng;
  title: string;
};

const mapPropsToTableRowValues = ({
  center,
  bounds,
}: Props): TableRowValue[] => [
  { latLng: center, title: "center" },
  { latLng: bounds.leftTop, title: "leftTop" },
  { latLng: bounds.rightBottom, title: "rightBottom" },
];

export function MeshDetail(props: Props): React.ReactElement<Props> {
  return (
    <Table inverted>
      <Tbody>
        <Tr>
          <Td>mesh code</Td>
          <Td>{props.code}</Td>
        </Tr>
        {mapPropsToTableRowValues(props).map(
          (value: TableRowValue, key: number) => (
            <Tr key={key}>
              <Td>{value.title}</Td>
              <Td>
                {round(value.latLng.lat, 5)}
                <br />
                {round(value.latLng.lng, 5)}
              </Td>
            </Tr>
          )
        )}
      </Tbody>
    </Table>
  );
}
