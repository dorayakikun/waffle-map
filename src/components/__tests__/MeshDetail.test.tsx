import * as React from "react";
import * as render from "react-test-renderer";
import { toBounds, toCenterLatLng } from "waffle-map-mesh-calculator-basic";
import { MeshDetail, Props } from "../MeshDetail";

test("Should set props to MeshDetail", () => {
  const meshCode = "5339";
  const center = toCenterLatLng(meshCode);
  const bounds = toBounds(meshCode);

  const props: Props = {
    bounds,
    center,
    code: meshCode
  };
  const tree = render.create(<MeshDetail {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
