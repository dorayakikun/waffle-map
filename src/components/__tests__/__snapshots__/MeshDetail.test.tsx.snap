// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should set props to MeshDetail 1`] = `
<table
  className="chakra-table emotion-0"
  inverted={true}
  role="table"
>
  <tbody
    className="emotion-0"
  >
    <tr
      className="emotion-0"
      role="row"
    >
      <td
        className="emotion-0"
        role="gridcell"
      >
        mesh code
      </td>
      <td
        className="emotion-0"
        role="gridcell"
      >
        5339
      </td>
    </tr>
    <tr
      className="emotion-0"
      role="row"
    >
      <td
        className="emotion-0"
        role="gridcell"
      >
        center
      </td>
      <td
        className="emotion-0"
        role="gridcell"
      >
        35.66667
        <br />
        139.5
      </td>
    </tr>
    <tr
      className="emotion-0"
      role="row"
    >
      <td
        className="emotion-0"
        role="gridcell"
      >
        leftTop
      </td>
      <td
        className="emotion-0"
        role="gridcell"
      >
        36
        <br />
        139
      </td>
    </tr>
    <tr
      className="emotion-0"
      role="row"
    >
      <td
        className="emotion-0"
        role="gridcell"
      >
        rightBottom
      </td>
      <td
        className="emotion-0"
        role="gridcell"
      >
        35.33334
        <br />
        140
      </td>
    </tr>
  </tbody>
</table>
`;
